cmd/vic-machine/converter/converter.go

```diff
@@ -0,0 +1,505 @@
+			delete(tempMap, k)
+		}
+	}
+	return nil
+}
+
+// keyAfterValueLabelHandler will add the map key as label after the value,
+// e.g. change from datastore/path to datastore/path:default
+func keyAfterValueLabelHandler(dest map[string][]string, pkey string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("map key: %s, map: %#v", pkey, dest)))
+
+	for _, values := range dest {
+		for i := range values {
+			values[i] = fmt.Sprintf("%s:%s", values[i], pkey)
+		}
+	}
+	return nil
+}
+
+// valueAfterKeyLabelHandler will add the map key as label before the value,
+// e.g. change from 10.10.10.0/24 to management:10.10.10.0/24
+func valueAfterKeyLabelHandler(dest map[string][]string, pkey string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("map key: %s, map: %#v", pkey, dest)))
+
+	for _, values := range dest {
+		for i := range values {
+			values[i] = fmt.Sprintf("%s:%s", pkey, values[i])
+		}
+	}
+	return nil
+}
+
+// calculateKey generate key as prefix-tag. if any one is empty, return the other
+func calculateKey(tags reflect.StructTag, prefix string) string {
+	tag := tags.Get(cmdTag)
+	if tag == "" {
+		return prefix
+	}
+	if tag == parentTagValue && prefix == "" {
+		return ""
+	}
+	if tag == parentTagValue && prefix != "" {
+		// for this tag, use parent name only
+		return prefix
+	}
+	if prefix == "" {
+		return tag
+	}
+	return fmt.Sprintf("%s%s%s", prefix, optionSeparator, tag)
+}
+
+func convertURL(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	if prefix == "" {
+		return nil
+	}
+	if tags.Get(cmdTag) == "" {
+		return nil
+	}
+
+	u, ok := src.Interface().(url.URL)
+	if !ok {
+		return fmt.Errorf(src.Type().String() + " is not URL")
+	}
+	v := u.String()
+	if u.Scheme == "" {
+		if u.Path == "" {
+			v = u.Host
+		} else if u.Host == "" {
+			v = u.Path
+		} else {
+			v = fmt.Sprintf("%s/%s", u.Host, u.Path)
+		}
+	}
+
+	log.Debugf("%s=%s", prefix, v)
+	addValue(dest, prefix, v)
+	return nil
+}
+
+func convertIPNet(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	if prefix == "" {
+		return nil
+	}
+	if tags.Get(cmdTag) == "" {
+		return nil
+	}
+
+	ipNet, ok := src.Interface().(net.IPNet)
+	if !ok {
+		return fmt.Errorf(src.Type().String() + " is not IPNet")
+	}
+	if ip.IsUnspecifiedSubnet(&ipNet) {
+		return nil
+	}
+	v := ipNet.String()
+
+	log.Debugf("%s=%s", prefix, v)
+	addValue(dest, prefix, v)
+	return nil
+}
+
+func convertIP(src reflect.Value, prefix string, tags reflect.StructTag, dest map[string][]string) error {
+	defer trace.End(trace.Begin(fmt.Sprintf("prefix: %s, src: %s", prefix, src.String())))
+
+	if prefix == "" {
+		return nil
+	}
+	if tags.Get(cmdTag) == "" {
+		return nil
+	}
+
+	ipAddr, ok := src.Interface().(net.IP)
```

Comment:
Taking into account you are using map of types. I would not be so tolerant politely returning error, I would panic here, since this situations is simply not possible.